circleci: failed tests: TestGossipPeerings TestDockerReadWriteForwardReferenceVersion

The following test appears to have failed:

[#19324](https://circleci.com/gh/cockroachdb/cockroach/19324):

```
I160620 17:03:16.135425 acceptance/replication_test.go:66  found 2 replicas
I160620 17:03:17.136825 acceptance/replication_test.go:66  found 2 replicas
I160620 17:03:17.237577 acceptance/cluster/localcluster.go:665  stopping
I160620 17:03:17.237826 acceptance/cluster/localcluster.go:563  event stream done, resetting...: net/http: request canceled
I160620 17:03:17.237843 acceptance/cluster/localcluster.go:586  events monitor exits
--- FAIL: TestGossipPeerings (19.01s)
	testing.go:113: acceptance/replication_test.go:72: condition failed to evaluate within 15s: expected 3 replicas, only found 2
=== RUN   TestGossipRestart
I160620 17:03:19.260121 acceptance/cluster/localcluster.go:295  Initializing Cluster AdHoc 3x1:
{"name":"AdHoc 3x1","nodes":[{"count":3,"stores":[{"count":1,"max_ranges":0}]}],"duration":5000000000}
I160620 17:03:19.262109 acceptance/cluster/docker.go:105  ImagePull cockroachdb/builder:20160611-170214 already exists
I160620 17:03:20.015875 acceptance/cluster/localcluster.go:597  creating certs (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.128850546
I160620 17:03:20.090529 acceptance/cluster/localcluster.go:548  events monitor starts
I160620 17:03:20.585728 acceptance/cluster/localcluster.go:497  *** started roach0 ***
  ui:        https://127.0.0.1:32823
  trace:     https://127.0.0.1:32823/debug/requests
--
dial tcp [::1]:26257: getsockopt: connection refused
Use the reference binary to write a couple rows, then render its output to a file and shut down.
I160620 17:05:17.472946 acceptance/cluster/localcluster.go:665  stopping
I160620 17:05:17.473213 acceptance/cluster/localcluster.go:563  event stream done, resetting...: net/http: request canceled
I160620 17:05:17.473232 acceptance/cluster/localcluster.go:586  events monitor exits
--- FAIL: TestDockerReadWriteForwardReferenceVersion (24.62s)
	reference_test.go:72: expected success: non-zero exit code: 1
=== RUN   TestRepair
--- SKIP: TestRepair (0.00s)
	repair_test.go:29: TODO(bram): skip this test until failures are investigated - #6798, #6700, #6277, #6209, #5672
=== RUN   TestDockerRuby
I160620 17:05:22.238469 acceptance/cluster/localcluster.go:295  Initializing Cluster ruby:
{"name":"ruby","nodes":[{"count":1,"stores":[{"count":1,"max_ranges":0}]}],"duration":5000000000}
I160620 17:05:22.240644 acceptance/cluster/docker.go:105  ImagePull cockroachdb/builder:20160611-170214 already exists
I160620 17:05:25.947994 acceptance/cluster/localcluster.go:597  creating certs (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.518651280
I160620 17:05:26.040807 acceptance/cluster/localcluster.go:548  events monitor starts
--
E160620 17:06:35.135786 acceptance/cluster/localcluster.go:513  node=2 status=restart
I160620 17:06:47.183450 acceptance/cluster/localcluster.go:665  stopping
I160620 17:06:47.183688 acceptance/cluster/localcluster.go:563  event stream done, resetting...: net/http: request canceled
I160620 17:06:47.183706 acceptance/cluster/localcluster.go:586  events monitor exits
--- PASS: TestRaftUpdate (33.76s)
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/storage/allocator.go
+++ b/storage/allocator.go
@@ -200,9 +200,18 @@ func (a *Allocator) AllocateTarget(required roachpb.Attributes, existing []roach
 	// matching here is lenient, and tries to find a target by relaxing an
 	// attribute constraint, from last attribute to first.
 	for attrs := append([]string(nil), required.Attrs...); ; attrs = attrs[:len(attrs)-1] {
-		sl, aliveStoreCount := a.storePool.getStoreList(roachpb.Attributes{Attrs: attrs}, a.options.Deterministic)
+		sl, aliveStoreCount, throttledStoreCount := a.storePool.getStoreList(
+			roachpb.Attributes{Attrs: attrs},
+			a.options.Deterministic,
+		)
 		if target := a.selectGood(sl, existingNodes); target != nil {
 			return target, nil
+		}
+
+		// When there are throttled stores that do match, we shouldn't send
+		// the replica to purgatory or even consider relaxing the constraints.
+		if throttledStoreCount > 0 {
+			return nil, util.Errorf("%d matching stores are currently throttled", throttledStoreCount)
 		}
 		if len(attrs) == 0 || !relaxConstraints {
 			return nil, &allocatorError{
@@ -274,7 +283,7 @@ func (a Allocator) RebalanceTarget(
 		existingNodes[repl.NodeID] = struct{}{}
 	}
 	storeDesc := a.storePool.getStoreDescriptor(storeID)
-	sl, _ := a.storePool.getStoreList(required, a.options.Deterministic)
+	sl, _, _ := a.storePool.getStoreList(required, a.options.Deterministic)
 	if replacement := a.improve(storeDesc, sl, existingNodes); replacement != nil {
 		return replacement
 	}
@@ -300,7 +309,7 @@ func (a *Allocator) ShouldRebalance(storeID roachpb.StoreID) bool {
 		return false
 	}
 
-	sl, _ := a.storePool.getStoreList(*storeDesc.CombinedAttrs(), a.options.Deterministic)
+	sl, _, _ := a.storePool.getStoreList(*storeDesc.CombinedAttrs(), a.options.Deterministic)
 
 	// ShouldRebalance is true if a suitable replacement can be found.
 	return a.improve(storeDesc, sl, makeNodeIDSet(storeDesc.Node.NodeID)) != nil

--- a/storage/allocator_test.go
+++ b/storage/allocator_test.go
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/config"
 	"github.com/cockroachdb/cockroach/gossip"
@@ -31,6 +32,7 @@ import (
 	"github.com/cockroachdb/cockroach/util/hlc"
 	"github.com/cockroachdb/cockroach/util/leaktest"
 	"github.com/cockroachdb/cockroach/util/stop"
+	"github.com/cockroachdb/cockroach/util/timeutil"
 )
 
 var simpleZoneConfig = config.ZoneConfig{
@@ -401,7 +403,7 @@ func TestAllocatorRebalance(t *testing.T) {
 		},
 		{
 			// This store will not be rebalanced to, because it already has more
-			// replias than the mean range count.
+			// replicas than the mean range count.
 			StoreID: 5,
 			Node:    roachpb.NodeDescriptor{NodeID: 5},
 			Capacity: roachpb.StoreCapacity{
@@ -1008,6 +1010,59 @@ func TestAllocatorError(t *testing.T) {
 	}
 }
 
+// TestAllocatorThrottled ensures that when a store is throttled, the replica
+// will not be sent to purgatory.
+func TestAllocatorThrottled(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	stopper, g, _, a, _ := createTestAllocator()
+	defer stopper.Stop()
+
+	// First test to make sure we would send the replica to purgatory.
+	_, err := a.AllocateTarget(
+		simpleZoneConfig.ReplicaAttrs[0],
+		[]roachpb.ReplicaDescriptor{},
+		false,
+		nil,
+	)
+	if _, ok := err.(purgatoryError); !ok {
+		t.Fatalf("expected a purgatory error, got: %v", err)
+	}
+
+	// Second, test the normal case in which we can allocate to the store.
+	gossiputil.NewStoreGossiper(g).GossipStores(singleStore, t)
+	result, err := a.AllocateTarget(
+		simpleZoneConfig.ReplicaAttrs[0],
+		[]roachpb.ReplicaDescriptor{},
+		false,
+		nil,
+	)
+	if err != nil {
+		t.Fatalf("unable to perform allocation: %v", err)
+	}
+	if result.Node.NodeID != 1 || result.StoreID != 1 {
+		t.Errorf("expected NodeID 1 and StoreID 1: %+v", result)
+	}
+
+	// Finally, set that store to be throttled and ensure we don't send the
+	// replica to purgatory.
+	a.storePool.mu.Lock()
+	storeDetail, ok := a.storePool.mu.stores[singleStore[0].StoreID]
+	if !ok {
+		t.Fatalf("store:%d was not found in the store pool", singleStore[0].StoreID)
+	}
+	storeDetail.throttledUntil = timeutil.Now().Add(24 * time.Hour)
+	a.storePool.mu.Unlock()
+	_, err = a.AllocateTarget(
+		simpleZoneConfig.ReplicaAttrs[0],
+		[]roachpb.ReplicaDescriptor{},
+		false,
+		nil,
+	)
+	if _, ok := err.(purgatoryError); ok {
+		t.Fatalf("expected a non purgatory error, got: %v", err)
+	}
+}
+
 type testStore struct {
 	roachpb.StoreDescriptor
 }
@@ -1087,7 +1142,7 @@ func Example_rebalancing() {
 			}
 		}
 
-		// Output store capacities as hexidecimal 2-character values.
+		// Output store capacities as hexadecimal 2-character values.
 		if i%(generations/50) == 0 {
 			var maxBytes int64
 			for j := 0; j < len(testStores); j++ {

--- a/storage/replicate_queue.go
+++ b/storage/replicate_queue.go
@@ -134,6 +134,10 @@ func (rq *replicateQueue) process(
 			NodeID:  newStore.Node.NodeID,
 			StoreID: newStore.StoreID,
 		}
+		if log.V(1) {
+			log.Infof("adding replica for under-replicated range:%d to node:%d, store:%d",
+				repl.RangeID, newReplica.NodeID, newReplica.StoreID)
+		}
 		if err = repl.ChangeReplicas(roachpb.ADD_REPLICA, newReplica, desc); err != nil {
 			return err
 		}
@@ -141,6 +145,10 @@ func (rq *replicateQueue) process(
 		removeReplica, err := rq.allocator.RemoveTarget(desc.Replicas)
 		if err != nil {
 			return err
+		}
+		if log.V(1) {
+			log.Infof("removing replica for over-replicated range:%d from node:%d, store:%d",
+				repl.RangeID, removeReplica.NodeID, removeReplica.StoreID)
 		}
 		if err = repl.ChangeReplicas(roachpb.REMOVE_REPLICA, removeReplica, desc); err != nil {
 			return err
@@ -156,6 +164,10 @@ func (rq *replicateQueue) process(
 			}
 			break
 		}
+		if log.V(1) {
+			log.Infof("removing replica from dead store range:%d from node:%d, store:%d",
+				repl.RangeID, deadReplicas[0].NodeID, deadReplicas[0].StoreID)
+		}
 		if err = repl.ChangeReplicas(roachpb.REMOVE_REPLICA, deadReplicas[0], desc); err != nil {
 			return err
 		}
@@ -164,6 +176,9 @@ func (rq *replicateQueue) process(
 		// rebalance. Attempt to find a rebalancing target.
 		rebalanceStore := rq.allocator.RebalanceTarget(repl.store.StoreID(), zone.ReplicaAttrs[0], desc.Replicas)
 		if rebalanceStore == nil {
+			if log.V(1) {
+				log.Infof("no suitable rebalance target for range:%d", repl.RangeID)
+			}
 			// No action was necessary and no rebalance target was found. Return
 			// without re-queuing this replica.
 			return nil
@@ -171,6 +186,10 @@ func (rq *replicateQueue) process(
 		rebalanceReplica := roachpb.ReplicaDescriptor{
 			NodeID:  rebalanceStore.Node.NodeID,
 			StoreID: rebalanceStore.StoreID,
+		}
+		if log.V(1) {
+			log.Infof("rebalancing range:%d to node:%d, store:%d", repl.RangeID,
+				rebalanceReplica.NodeID, rebalanceReplica.StoreID)
 		}
 		if err = repl.ChangeReplicas(roachpb.ADD_REPLICA, rebalanceReplica, desc); err != nil {
 			return err

--- a/storage/replicate_queue.go
+++ b/storage/replicate_queue.go
@@ -135,8 +135,7 @@ func (rq *replicateQueue) process(
 			StoreID: newStore.StoreID,
 		}
 		if log.V(1) {
-			log.Infof("adding replica for under-replicated range:%d to node:%d, store:%d",
-				repl.RangeID, newReplica.NodeID, newReplica.StoreID)
+			log.Infof("adding replica for under-replicated RangeID:%d to %+v", repl.RangeID, newReplica)
 		}
 		if err = repl.ChangeReplicas(roachpb.ADD_REPLICA, newReplica, desc); err != nil {
 			return err
@@ -147,8 +146,7 @@ func (rq *replicateQueue) process(
 			return err
 		}
 		if log.V(1) {
-			log.Infof("removing replica for over-replicated range:%d from node:%d, store:%d",
-				repl.RangeID, removeReplica.NodeID, removeReplica.StoreID)
+			log.Infof("removing replica for over-replicated RangeID:%d from %+v", repl.RangeID, removeReplica)
 		}
 		if err = repl.ChangeReplicas(roachpb.REMOVE_REPLICA, removeReplica, desc); err != nil {
 			return err
@@ -165,8 +163,7 @@ func (rq *replicateQueue) process(
 			break
 		}
 		if log.V(1) {
-			log.Infof("removing replica from dead store range:%d from node:%d, store:%d",
-				repl.RangeID, deadReplicas[0].NodeID, deadReplicas[0].StoreID)
+			log.Infof("removing replica from dead store RangeID:%d from %+v", repl.RangeID, deadReplicas[0])
 		}
 		if err = repl.ChangeReplicas(roachpb.REMOVE_REPLICA, deadReplicas[0], desc); err != nil {
 			return err
@@ -177,7 +174,7 @@ func (rq *replicateQueue) process(
 		rebalanceStore := rq.allocator.RebalanceTarget(repl.store.StoreID(), zone.ReplicaAttrs[0], desc.Replicas)
 		if rebalanceStore == nil {
 			if log.V(1) {
-				log.Infof("no suitable rebalance target for range:%d", repl.RangeID)
+				log.Infof("no suitable rebalance target for RangeID:%d", repl.RangeID)
 			}
 			// No action was necessary and no rebalance target was found. Return
 			// without re-queuing this replica.
@@ -188,8 +185,7 @@ func (rq *replicateQueue) process(
 			StoreID: rebalanceStore.StoreID,
 		}
 		if log.V(1) {
-			log.Infof("rebalancing range:%d to node:%d, store:%d", repl.RangeID,
-				rebalanceReplica.NodeID, rebalanceReplica.StoreID)
+			log.Infof("rebalancing RangeID:%d to %+v", repl.RangeID, rebalanceReplica)
 		}
 		if err = repl.ChangeReplicas(roachpb.ADD_REPLICA, rebalanceReplica, desc); err != nil {
 			return err

--- a/storage/store_pool.go
+++ b/storage/store_pool.go
@@ -45,11 +45,11 @@ const (
 	TestTimeUntilStoreDeadOff = 24 * time.Hour
 
 	// defaultFailedReservationsTimeout is the amount of time to consider the
-	// store unavailable for up-replication after a failed reservation call.
+	// store throttled for up-replication after a failed reservation call.
 	defaultFailedReservationsTimeout = 5 * time.Second
 
 	// defaultDeclinedReservationsTimeout is the amount of time to consider the
-	// store unavailable for up-replication after a reservation was declined.
+	// store throttled for up-replication after a reservation was declined.
 	defaultDeclinedReservationsTimeout = 0 * time.Second
 
 	// defaultReserveRPCTimeout is used for the rpc calls to Reserve on other
@@ -62,11 +62,11 @@ type storeDetail struct {
 	dead        bool
 	timesDied   int
 	foundDeadOn hlc.Timestamp
-	// unavailableUntil is when an unavailable store can be considered available
+	// throttledUntil is when an throttled store can be considered available
 	// again due to a failed or declined Reserve RPC.
-	unavailableUntil time.Time
-	lastUpdatedTime  hlc.Timestamp // This is also the priority for the queue.
-	index            int           // index of the item in the heap, required for heap.Interface
+	throttledUntil  time.Time
+	lastUpdatedTime hlc.Timestamp // This is also the priority for the queue.
+	index           int           // index of the item in the heap, required for heap.Interface
 }
 
 // markDead sets the storeDetail to dead(inactive).
@@ -94,25 +94,31 @@ type storeMatch int
 
 // These are the possible values for a storeMatch.
 const (
-	storeMatchDead    storeMatch = iota // The store is not yet available or has been timed out.
-	storeMatchAlive                     // The store is alive, but its attributes didn't match the required ones.
-	storeMatchMatched                   // The store is alive and its attributes matched.
+	storeMatchDead      storeMatch = iota // The store is not yet available or has been timed out.
+	storeMatchAlive                       // The store is alive, but its attributes didn't match the required ones.
+	storeMatchThrottled                   // The store is alive and its attributes matched, but it is throttled.
+	storeMatchAvailable                   // The store is alive, available and its attributes matched.
 )
 
-// match returns if the store is alive, and if the store is available and it's
-// attributes contain the required ones respectively.
+// match checks the store against the attributes and returns a storeMatch.
 func (sd *storeDetail) match(now time.Time, required roachpb.Attributes) storeMatch {
 	// The store must be alive and it must have a descriptor to be considered
 	// alive.
 	if sd.dead || sd.desc == nil {
 		return storeMatchDead
 	}
-	// The store must not have a recent declined reservation to be considered
-	// available for matching.
-	if sd.unavailableUntil.After(now) || !required.IsSubset(*sd.desc.CombinedAttrs()) {
+
+	// Does the store match the attributes?
+	if !required.IsSubset(*sd.desc.CombinedAttrs()) {
 		return storeMatchAlive
 	}
-	return storeMatchMatched
+
+	// The store must not have a recent declined reservation to be available.
+	if sd.throttledUntil.After(now) {
+		return storeMatchThrottled
+	}
+
+	return storeMatchAvailable
 }
 
 // storePoolPQ implements the heap.Interface (which includes sort.Interface)
@@ -381,11 +387,11 @@ func (sl *StoreList) add(s *roachpb.StoreDescriptor) {
 
 // GetStoreList returns a storeList that contains all active stores that
 // contain the required attributes and their associated stats. It also returns
-// the number of total alive stores.
+// the total number of alive and throttled stores.
 // TODO(embark, spencer): consider using a reverse index map from
 // Attr->stores, for efficiency. Ensure that entries in this map still
 // have an opportunity to be garbage collected.
-func (sp *StorePool) getStoreList(required roachpb.Attributes, deterministic bool) (StoreList, int) {
+func (sp *StorePool) getStoreList(required roachpb.Attributes, deterministic bool) (StoreList, int, int) {
 	sp.mu.RLock()
 	defer sp.mu.RUnlock()
 
@@ -401,17 +407,22 @@ func (sp *StorePool) getStoreList(required roachpb.Attributes, deterministic boo
 	now := sp.clock.Now().GoTime()
 	sl := StoreList{}
 	var aliveStoreCount int
+	var throttledStoreCount int
 	for _, storeID := range storeIDs {
 		detail := sp.mu.stores[roachpb.StoreID(storeID)]
 		matched := detail.match(now, required)
-		if matched >= storeMatchAlive {
+		switch matched {
+		case storeMatchAlive:
 			aliveStoreCount++
-		}
-		if matched == storeMatchMatched {
+		case storeMatchThrottled:
+			aliveStoreCount++
+			throttledStoreCount++
+		case storeMatchAvailable:
+			aliveStoreCount++
 			sl.add(detail.desc)
 		}
 	}
-	return sl, aliveStoreCount
+	return sl, aliveStoreCount, throttledStoreCount
 }
 
 // reserve send a reservation request rpc to the node and store
@@ -466,18 +477,18 @@ func (sp *StorePool) reserve(
 	// be considered as a candidate for new replicas until after the configured
 	// timeout period has passed.
 	if err != nil {
-		detail.unavailableUntil = sp.clock.Now().GoTime().Add(sp.failedReservationsTimeout)
+		detail.throttledUntil = sp.clock.Now().GoTime().Add(sp.failedReservationsTimeout)
 		if log.V(2) {
-			log.Infof("reservation failed, store:%s will be unavailable for %s until %s",
-				toStoreID, sp.failedReservationsTimeout, detail.unavailableUntil)
+			log.Infof("reservation failed, store:%s will be throttled for %s until %s",
+				toStoreID, sp.failedReservationsTimeout, detail.throttledUntil)
 		}
 		return fmt.Errorf("reservation failed:%+v due to error:%s", req, err)
 	}
 	if !resp.Reserved {
-		detail.unavailableUntil = sp.clock.Now().GoTime().Add(sp.declinedReservationsTimeout)
+		detail.throttledUntil = sp.clock.Now().GoTime().Add(sp.declinedReservationsTimeout)
 		if log.V(2) {
-			log.Infof("reservation failed, store:%s will be unavailable for %s until %s",
-				toStoreID, sp.declinedReservationsTimeout, detail.unavailableUntil)
+			log.Infof("reservation failed, store:%s will be throttled for %s until %s",
+				toStoreID, sp.declinedReservationsTimeout, detail.throttledUntil)
 		}
 		return fmt.Errorf("reservation declined:%+v", req)
 	}

--- a/storage/store_pool_test.go
+++ b/storage/store_pool_test.go
@@ -18,7 +18,6 @@ package storage
 
 import (
 	"errors"
-	"fmt"
 	"reflect"
 	"sort"
 	"testing"
@@ -209,12 +208,22 @@ func TestStorePoolDies(t *testing.T) {
 }
 
 // verifyStoreList ensures that the returned list of stores is correct.
-func verifyStoreList(sp *StorePool, requiredAttrs []string, expected []int, expectedAliveStoreCount int) error {
+func verifyStoreList(
+	sp *StorePool,
+	requiredAttrs []string,
+	expected []int,
+	expectedAliveStoreCount int,
+	expectedThrottledStoreCount int,
+) error {
 	var actual []int
-	sl, aliveStoreCount := sp.getStoreList(roachpb.Attributes{Attrs: requiredAttrs}, false)
+	sl, aliveStoreCount, throttledStoreCount := sp.getStoreList(roachpb.Attributes{Attrs: requiredAttrs}, false)
 	if aliveStoreCount != expectedAliveStoreCount {
-		return fmt.Errorf("expected AliveStoreCount %d does not match actual %d", expectedAliveStoreCount,
-			aliveStoreCount)
+		return util.Errorf("expected AliveStoreCount %d does not match actual %d",
+			expectedAliveStoreCount, aliveStoreCount)
+	}
+	if throttledStoreCount != expectedThrottledStoreCount {
+		return util.Errorf("expected ThrottledStoreCount %d does not match actual %d",
+			expectedThrottledStoreCount, throttledStoreCount)
 	}
 	for _, store := range sl.stores {
 		actual = append(actual, int(store.StoreID))
@@ -222,7 +231,7 @@ func verifyStoreList(sp *StorePool, requiredAttrs []string, expected []int, expe
 	sort.Ints(expected)
 	sort.Ints(actual)
 	if !reflect.DeepEqual(expected, actual) {
-		return fmt.Errorf("expected %+v stores, actual %+v", expected, actual)
+		return util.Errorf("expected %+v stores, actual %+v", expected, actual)
 	}
 	return nil
 }
@@ -237,7 +246,7 @@ func TestStorePoolGetStoreList(t *testing.T) {
 	sg := gossiputil.NewStoreGossiper(g)
 	required := []string{"ssd", "dc"}
 	// Nothing yet.
-	if sl, _ := sp.getStoreList(roachpb.Attributes{Attrs: required}, false); len(sl.stores) != 0 {
+	if sl, _, _ := sp.getStoreList(roachpb.Attributes{Attrs: required}, false); len(sl.stores) != 0 {
 		t.Errorf("expected no stores, instead %+v", sl.stores)
 	}
 
@@ -287,20 +296,20 @@ func TestStorePoolGetStoreList(t *testing.T) {
 		int(supersetStore.StoreID),
 		int(deadStore.StoreID),
 		int(declinedStore.StoreID),
-	}, 6); err != nil {
+	}, 6, 0); err != nil {
 		t.Error(err)
 	}
 
 	// Mark one store dead and one store declined.
 	sp.mu.Lock()
 	sp.mu.stores[deadStore.StoreID].markDead(sp.clock.Now())
-	sp.mu.stores[declinedStore.StoreID].unavailableUntil = sp.clock.Now().GoTime().Add(time.Hour)
+	sp.mu.stores[declinedStore.StoreID].throttledUntil = sp.clock.Now().GoTime().Add(time.Hour)
 	sp.mu.Unlock()
 
 	if err := verifyStoreList(sp, required, []int{
 		int(matchingStore.StoreID),
 		int(supersetStore.StoreID),
-	}, 5); err != nil {
+	}, 5, 1); err != nil {
 		t.Error(err)
 	}
 }
@@ -414,8 +423,15 @@ func TestStorePoolDefaultState(t *testing.T) {
 		t.Errorf("expected 0 dead replicas; got %v", dead)
 	}
 
-	if sl, c := sp.getStoreList(roachpb.Attributes{}, true); len(sl.stores) > 0 || c != 0 {
-		t.Errorf("expected 0 live stores; got list %v and total count %d", sl, c)
+	sl, alive, throttled := sp.getStoreList(roachpb.Attributes{}, true)
+	if len(sl.stores) > 0 {
+		t.Errorf("expected no live stores; got list of %v", sl)
+	}
+	if alive != 0 {
+		t.Errorf("expected no live stores; got an alive count of %d", alive)
+	}
+	if throttled != 0 {
+		t.Errorf("expected no live stores; got a throttled count of %d", throttled)
 	}
 }
 
@@ -518,7 +534,7 @@ func TestStorePoolReserve(t *testing.T) {
 	for i, testCase := range testCases {
 		f.reservationResponse = testCase.fakeResp
 		if len(testCase.fakeErr) != 0 {
-			f.reservationErr = fmt.Errorf("%s", testCase.fakeErr)
+			f.reservationErr = util.Errorf("%s", testCase.fakeErr)
 		} else {
 			f.reservationErr = nil
 		}

